File: Library.Core.Web/src/app/view-models/author/postAuthorView.ts
export class PostAuthorView {
  constructor(
    public id: number = 0,
    public name?: string,
    public birthday: number = 0,
    public deathday: number = 0,
    public books?: BookPostAuthorViewItem[]
  ) { }
}

export class BookPostAuthorViewItem {
  constructor(
    public id: number = 0,
    public name?: string,
    public yearOfPublishing: number = 0
  ) { }
}

File: Library.Core.ViewModelLayer/ViewModels/AuthorViewModels/PostAuthorView.cs
using System.Collections.Generic;

namespace Library.Core.ViewModelLayer.ViewModels.Author
{
    public class PostAuthorView
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Birthday { get; set; }
        public int Deathday { get; set; }

        public ICollection<BookPostAuthorViewItem> Books { get; set; }

        public PostAuthorView()
        {
            Books = new List<BookPostAuthorViewItem>();
        }
    }

    public class BookPostAuthorViewItem
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int YearOfPublishing { get; set; }

        public BookPostAuthorViewItem()
        {

        }
    }
}
